### PR TITLE
Sites dashboard v2 - Add/status filter functionality to v2 sites dashboard

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
Related to # https://github.com/Automattic/dotcom-forge/issues/6711

## Proposed Changes

* Adds filtering functionality to the v2 sites dashboard.
* Note - pagination, sorting, etc. are not yet connected and scoped for follow up PRs.

BEFORE
Screenshot 2024-04-24 at 2 48 54 PM

AFTER
Screenshot 2024-04-24 at 2 49 27 PM

## Testing Instructions

* Run this calypso branch.
* Visit the sites dashboard */sites with the feature flag ?flags=layout/dotcom-nav-redesign-v2 added to the url.
* Test the filters dropdown in the sites dashboard and select statuses.
* Verify the sites list updates and shows only sites of the selected status.
* Verify the selected status as been added to the URL as a query param.
* Reload the browser window, the sites dashboard should initialize with the status from the URL and the list should show only sites of this status as expect.
* Test alongside search input queries as well.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/sites-dashboard-v2/sites-dataviews/index.tsx [hunk 1]
```diff
@@ -26,6 +27,15 @@ type Props = {
...
 };
 
+export const statuses = [
+	{ value: 1, label: __( 'All sites' ), slug: 'all' },
+	{ value: 2, label: __( 'Public' ), slug: 'public' },
+	{ value: 3, label: __( 'Private' ), slug: 'private' },
+	{ value: 4, label: __( 'Coming soon' ), slug: 'coming-soon' },
+	{ value: 5, label: __( 'Redirect' ), slug: 'redirect' },
+	{ value: 6, label: __( 'Deleted' ), slug: 'deleted' },
+];
```
Comment (Addison-Stavlo) [inline client/sites-dashboard-v2/sites-dataviews/index.tsx]: I moved these out of the component to make them an accessible export and tie them to the slugs we use for filtering on these values. This gives us a more direct connection between integer values used by the data component and our slugs for filtering.

**Review by davemart-in — APPROVED** [review]
Followed your testing instructions. Everything works as expected. No bugs.

File: client/sites-dashboard-v2/sites-dataviews/index.tsx (hunk 1 above)
Comment (cleacos) [reply client/sites-dashboard-v2/sites-dataviews/index.tsx]: Great. At A4A, we created a map to target those integer values:

`client/a8c-for-agencies/sections/sites/constants.ts`

```
export const filtersMap: AgencyDashboardFilterMap[] = [
	{ filterType: 'all_issues', ref: 1 },
	{ filterType: 'backup_failed', ref: 2 },
	{ filterType: 'backup_warning', ref: 3 },
	{ filterType: 'threats_found', ref: 4 },
	{ filterType: 'site_disconnected', ref: 5 },
	{ filterType: 'site_down', ref: 6 },
	{ filterType: 'plugin_updates', ref: 7 },
];
```

And then we do things like this:

```
return {
	field: 'status',
	operator: 'in',
	value: filtersMap.find( ( filterMap ) => filterMap.filterType === issueType )?.ref || 1,
};
```

But we are still improving things there.

File: client/sites-dashboard-v2/index.tsx [hunk 2]
```diff
@@ -87,24 +118,16 @@ const SitesDashboardV2 = ( {
...
-	// Search, filtering, pagination and sorting sites:
-	useEffect( () => {
-		// todo: Perform search, filter, pagination and sorting actions
-	}, [
-		dataViewsState.search,
-		dataViewsState.filters,
-		dataViewsState.sort,
-		dataViewsState.page,
-		dataViewsState.perPage,
-	] );
```
Comment (cleacos) [inline client/sites-dashboard-v2/index.tsx:107]: ❤️

**Review by cleacos — APPROVED** [review]
It works perfectly!

LGTM!

**Comment by Addison-Stavlo** [comment]
Thanks for the reviews!  I just pushed one commit to change a variable name I found too vague in retrospect. Will send this out once checks are good.

**Pull request merged**